From c0f2d123ce16ef5b14347090c89f45606f5be1e5 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:53 -0400
Subject: [ethernet] qlcnic: Fix SR-IOV cleanup code path

Message-id: <1408626422-3941-70-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88601
O-Subject: [PATCH RHEL 7.1 069/138] qlcnic: Fix SR-IOV cleanup code path
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Manish Chopra <manish.chopra@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 6e1f586d31ad49063da391db12632b31c7b00d76

o Add __QLCNIC_SRIOV_ENABLE bit check before doing SRIOV cleanup

Signed-off-by: Manish Chopra <manish.chopra@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
index bf8fca7..f998fdc 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_common.c
@@ -277,9 +277,7 @@ static void qlcnic_sriov_vf_cleanup(struct qlcnic_adapter *adapter)
 
 void qlcnic_sriov_cleanup(struct qlcnic_adapter *adapter)
 {
- struct qlcnic_sriov *sriov = adapter->ahw->sriov;
-
- if (!sriov)
+ if (!test_bit(__QLCNIC_SRIOV_ENABLE, &adapter->state))
   return;
 
  qlcnic_sriov_free_vlans(adapter);
-- 
1.7.1